src/scope-manager.js

```diff
@@ -446,6 +459,10 @@ var scopeManager = function(state, predefined, exported, declared) {
           "(type)" : type,
           "(token)": token,
           "(unused)": true };
+
+        if (!_current["(params)"]) {
+          _current["(params)"] = [];
```

Comment:
> This seems like initialization logic. Does it really need to be dynamic? Since
> the _newScope function now receives the block type, is there any chance we
> can extend the logic there to create the array where necessary? My goal here
> is to make that function more descriptive of the "shape" of the scope object.
> As it stands, it's a little cagey about the "(params)" attribute (both its
> type and when it may be used). What do you think of:

Yes, thats better. I'm sure I had it like this and changed it, I think I was trying to minimise the number of magic values to stack and make it more flexible - but yes wanting to know the shape of the object was why I put (params) in set to null, this is fine.